src/articles/articles.service.ts

```diff
@@ -0,0 +1,142 @@
+import { Article, User } from './../../generated/prisma/index.d';
+import { BadRequestException, ConflictException, Injectable } from '@nestjs/common';
+import { CreateArticleDto } from './dto/create-article.dto';
+import { UpdateArticleDto } from './dto/update-article.dto';
+import slugify from 'slugify';
+import { PrismaService } from '../../prisma/prisma.service';
+
+@Injectable()
+export class ArticlesService {
+  constructor(private prisma: PrismaService) {}
+  async create(currUser: User, createArticleDto: CreateArticleDto) {
+    const slug = slugify(createArticleDto.title, {
+      lower: true,
+      strict: true,
+    });
+
+    const existedArticle = await this.prisma.article.findUnique({
+      where: { slug: slug },
+    });
+
+    if (existedArticle) {
+      throw new ConflictException(`Article with title "${createArticleDto.title}" already exists`);
+    }
+
+    const author = await this.prisma.user.findUnique({
+      where: { username: currUser.username },
+    });
+
+    if (!author) {
+      throw new Error('User not found');
+    }
+
+    let tagListData: { connect: { id: number }[]; create: { name: string }[] } | undefined;
```

Comment:
```suggestion
    let tagListData: <what is this?> | undefined;
```

em define thành 1 type riêng, đặt tên cho dễ hiểu nhé.
viết trực tiếp như này hơi cồng kềnh